### ISSUE TITLE
(Feature request) Support default background color in package.json

### ISSUE THREAD

**Issue by AshleyScirra** [nwjs/nw.js#5502]
For a nice bit of extra polish, it would be good if we could provide a default background color for the NW.js window to be used *before* the first page has loaded. Currently we have two options:

1) let window show immediately, which can show a flash of white, then the content loads - distracting if the page uses a dark background, e.g. black
2) specify a hidden window in package.json, and show it when the page is ready - but this adds some extra lag to the time until the window first appears

If we can specify a color in package.json then we get the best of both worlds: the window appears ASAP, and there is not a distracting switch.

Web apps use a similar technique by allowing a custom splash screen color in the web app manifest.

**Comment by ghostoy** [comment]
Default background color won't help. A complete blank window on start for a long time, no matter it's white or black, looks bad.

Actually you won't rely on a new feature of NW.js. You can simply delay the showing of a window until it's fully loaded. For example, mark your main window as hidden in `package.json` and popup a window holding splash screen. Show the main window when it's fully loaded and close the splash window. It won't bring in much overhead.

**Comment by AshleyScirra** [comment]
I disagree, it would help with perceived performance. It's a common technique for making loading *seem* faster (even if it is not actually faster). For example Facebook shows fake placeholder images of news items while waiting for your news feed to load, which improves the perceived performance.

Showing our own splash screen means nothing happens until the app has loaded far enough to execute JS, which can be a few seconds on some systems, particularly because NW.js has a fairly large overhead.

**Comment by ghostoy** [comment]
I didn't deny the usage of splash window. I was saying you can do it with NW.js without any new feature.

A splash is merely a frameless window containing an image. It can be done by NW.js currently. With NW.js, only two files are loaded from disk and rendered: an HTML and an image. If you found it slow, please report an issue and we will fix it.

JS is executed immediately after it's loaded in HTML unless `async` is set. It won't be much time until your JS can open a splash window if you make it the first script in your main HTML. If it's even not fast enough for you, you can use a `js` file as `main` and open the splash window in it followed by opening the hidden main window.

Example: `package.json`

```json
{
  "name": "your-app",
  "main": "main.js",
  "window": {
    "show": false
  }
}
```

`main.js`

```javascript
// open a frameless splash window
nw.Window.open('splash.html', {frame: false, width: 400, height:300});
// open a hidden main window
// and in index.html, it's responsible to show the window when it's fully loaded
nw.Window.open('index.html', {show: false});
```

**Comment by rogerwang** [comment]
This feature is useful when you have an app with dark UI, but when the window is resized/growing fast you'll see white rectangles in the new area before the re-layout is complete.

**Comment by fujunwei** [comment]
I'm working on supporting this feature, @AshleyScirra could you please share your test case to produce this issue.

**Comment by rogerwang** [comment]
@fujunwei please see #2837 . You can also start with a web page with dark background, resize the window with mouse fast, then you'll see the white, un-paint area in the growing window.

**Comment by rogerwang** [comment]
@fujunwei you can also refer to the code supporting the transparency feature to see how it sets a default pixel value for transparent background.

**Comment by fujunwei** [comment]
@rogerwang It still can't be reproduced on my Ubuntu 14.04 and Windows 10 machine, i will face to face with you in your free time.

**Comment by vankasteelj** [comment]
If it can help, I had the same issue but used this, and it's way better even if the app doesn't appear for a few milliseconds. 

With `{"window": {"show": false}}` in `package.json`:

```js
const gui = require('nw.gui');
gui.start = () => {
   return new Promise((resolve, reject) => {
       // load my things
       resolve();
    });
};

console.time('Application ready');
gui.start().then(() => {
    win.show(true);
    console.timeEnd('Application ready');
});
```
